testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1179,9 +1190,34 @@ public void testTrinoWritingDataWithWriterDataPathSet(StorageFormat storageForma
         assertThat(queryResult).hasRowsCount(1).hasColumnsCount(1);
         assertThat(((String) queryResult.getOnlyValue())).contains(dataPath);
 
-        assertQueryFailure(() -> onTrino().executeQuery("DROP TABLE " + trinoTableName))
-                .hasMessageContaining("contains Iceberg path override properties and cannot be dropped from Trino");
-        onSpark().executeQuery("DROP TABLE " + sparkTableName);
+        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+    }
+
+    @Test(groups = {ICEBERG, ICEBERG_JDBC, PROFILE_SPECIFIC_TESTS, ICEBERG_NESSIE}, dataProvider = "storageFormatsWithSpecVersion")
+    public void testSparkReadingTrinoObjectStorage(StorageFormat storageFormat, int specVersion)
+    {
+        String baseTableName = toLowerCase("test_trino_object_storage_location_provider_" + storageFormat);
+        String sparkTableName = sparkTableName(baseTableName);
+        String trinoTableName = trinoTableName(baseTableName);
+        String dataPath = "hdfs://hadoop-master:9000/user/hive/warehouse/test_trino_object_storage_location_provider/obj-data";
+
+        onTrino().executeQuery(format("CREATE TABLE %s (_string VARCHAR, _bigint BIGINT) WITH (" +
+                          "object_store_enabled = true," +
+                          "data_location = '%s'," +
+                          "format = '%s'," +
+                          "format_version = %s)",
+                  trinoTableName, dataPath, storageFormat, specVersion));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES ('a_string', 1000000000000000)", trinoTableName));
+
+        Row result = row("a_string", 1000000000000000L);
+        assertThat(onSpark().executeQuery(format("SELECT _string, _bigint FROM %s", sparkTableName))).containsOnly(result);
+        assertThat(onTrino().executeQuery(format("SELECT _string, _bigint FROM %s", trinoTableName))).containsOnly(result);
+
+        QueryResult queryResult = onTrino().executeQuery(format("SELECT file_path FROM %s", trinoTableName("\"" + baseTableName + "$files\"")));
+        assertThat(queryResult).hasRowsCount(1).hasColumnsCount(1);
+        assertThat(((String) queryResult.getOnlyValue()).contains(dataPath)).isTrue();
```

Comment:
Please avoid boolean assertion as much as possible. This can be changed to `assertThat(((String) queryResult.getOnlyValue())).contains(dataPath)`